chain/client/src/archive/cloud_archival_actor.rs

```diff
@@ -46,26 +45,21 @@ pub struct CloudArchivalActor {
     handle: CloudArchivalHandle,
 }
 
-/// Creates the cloud archival actor with handle if cloud archival writer is configured.
+/// Creates the cloud archival actor if cloud archival writer is configured.
 pub fn create_cloud_archival_actor(
     cloud_archival_config: Option<CloudArchivalWriterConfig>,
     genesis_height: BlockHeight,
-    storage: &NodeStorage,
-) -> anyhow::Result<Option<(CloudArchivalActor, CloudArchivalHandle)>> {
+    hot_store: Store,
+) -> anyhow::Result<Option<CloudArchivalActor>> {
     let Some(config) = cloud_archival_config else {
-        tracing::debug!(target : "cloud_archival", "Not creating the cloud archival actor because cloud archival writer is not configured");
+        tracing::debug!(target: "cloud_archival", "Not creating the cloud archival actor because cloud archival writer is not configured");
         return Ok(None);
     };
-    let hot_store = storage.get_hot_store();
-
-    // TODO(cloud_archival) Retrieve the `cloud_head` from the external storage
-    let cloud_head = hot_final_head_height(&hot_store, genesis_height)?;
+    let cloud_head = get_hot_final_head_height(&hot_store, genesis_height)?;
 
```

Comment:
It's still reading it from the local storage isn't it? 

Btw "what is the cloud head?" is an interesting question in the multi-writer + redundancy architecture, maybe a good topic for discussion?

Reply:
I've removed this TODO by mistake. It is removed in the next PR, will bring it back here.

> what is the cloud head?

I’ve done some thinking on it and agree it’s worth a discussion.